react-components/src/components/RuleBasedOutputs/RuleBasedOutputsPanel.tsx

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import styled from 'styled-components';
+
+import { Flex, Button } from '@cognite/cogs.js';
+import { type RuleOutput } from './types';
+import { useState, type ReactNode, type ReactElement } from 'react';
+import { withSuppressRevealEvents } from '../../higher-order-components/withSuppressRevealEvents';
+import { useTranslation } from '../i18n/I18n';
+
+const MAX_WHEN_NO_SHOW_MORE = 5;
+const MAX_HEIGHT = '300px';
+
+type RuleBasedOutputsPanelProps = {
+  ruleOutputs?: RuleOutput[];
+};
+
+export const RuleBasedOutputsPanel = ({
+  ruleOutputs
+}: RuleBasedOutputsPanelProps): ReactElement => {
+  const [isShowMore, setIsShowMore] = useState(false);
+
+  const { t } = useTranslation();
+
+  return (
+    <>
+      {ruleOutputs !== undefined && ruleOutputs?.length > 0 && (
+        <StyledRuleBasedOutputsPanel $showMore={isShowMore}>
+          <Flex direction="column" gap={14}>
+            {ruleOutputs.map((output, index) => {
+              return (
+                <>
+                  {isShowMore || index <= MAX_WHEN_NO_SHOW_MORE - 1 ? (
+                    <StyledOutputItem key={index}>
+                      <StyledBulletColor
+                        $backgroundColor={output.type === 'color' ? output.fill : ''}
+                      />
+                      <StyledLabel>
+                        {output.label ?? (output.type === 'color' ? output.fill : '')}
+                      </StyledLabel>
+                    </StyledOutputItem>
+                  ) : (
+                    <></>
+                  )}
+                </>
+              );
+            })}
+            {ruleOutputs.length > MAX_WHEN_NO_SHOW_MORE && (
+              <StyledOutputItem key={'rule-based-panel-show-more'}>
+                <ShowMore
+                  isShowMore={isShowMore}
+                  onShowMore={() => {
+                    setIsShowMore((previousState) => !previousState);
+                  }}>
+                  {isShowMore ? t('SHOW_LESS', 'Show less') : t('SHOW_MORE', 'Show more')}
+                </ShowMore>
+              </StyledOutputItem>
+            )}
+          </Flex>
+        </StyledRuleBasedOutputsPanel>
+      )}
+    </>
```

Comment:
This was a pretty rough block with many levels of indentation :sweat_smile: Could we factor out more smaller components here? E.g. a component for the StyledBulletColor+StyledLabel?

Reply:
yep

https://github.com/cognitedata/reveal/pull/4453/commits/4211e566168fb77e7af2c109a5b0d385d9d98fc4